### ISSUE TITLE
False positive for math sqrt function

### ISSUE THREAD

**Issue by ravisumit33** [PyCQA/pylint#4592]
### Steps to reproduce

Given a file `a.py`:
```python
"""
Module to test pylint
"""
from math import sqrt
print(sqrt(4))
```

### Current behavior

Result of `pylint a.py`:
<img width="762" alt="Screenshot 2021-06-18 at 6 55 54 PM" src="https://user-images.githubusercontent.com/19284116/122567923-d2fd7680-d066-11eb-9bf7-f9eb4936ab06.png">

### Expected behavior
pylint should not give error for math.sqrt function

### pylint --version output

Result of `pylint --version` output:

On Stable release
```
pylint 2.8.3
astroid 2.5.6
Python 3.9.5 (default, May  4 2021, 03:33:11)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

On preview release
```
pylint 3.0.0a3
astroid 2.6.0-dev0
Python 3.9.5 (default, May  4 2021, 03:33:11)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for opening the issue, I can't reproduce this problem. What is your operating system and environment ?

**Comment by ravisumit33** [comment]
I am working on Mac OS Catalina in terminal. Install pylint using `pip3 install pylint` and got this error.

**Comment by ravisumit33** [comment]
This duplicates https://github.com/PyCQA/pylint/issues/1524
I solved it using `extension-pkg-allow-list=math` in my pylintrc. However, I didn't get `c-extension-no-member` error rather `no-name-in-module` error which created the confusion.

**Comment by Pierre-Sassoulas** [comment]
Thank you for solving this. You must have a C binding for ``math`` on MacOs. I think I never saw the ``c-extension-no-member`` myself, half the issue about false positives for no-member and half the questions on stackoverflow could be avoided if ``c-extension-no-member`` was working properly because very often the answer is to put it in the ``extension-pkg-allow-list``.
